cvcl/urls.py

```diff
@@ -28,4 +29,5 @@
     url(r'^envdefs/(?P<pk>\d+)/createvmdef$', VmDefinitionCreate.as_view(), name='envdefs.createvmdef'),  # instructor
     url(r'^vmdefinitions/(?P<pk>\d+)/update$', VmDefinitionUpdate.as_view(), name='vmdef.update'),  # instructor
     url(r'^vmdefinitions/(?P<pk>\d+)/delete$', VmDefinitionDelete.as_view(), name='vmdef.delete'),  # instructor
-]
+                  url(r'^upload/', addreport, name='csv_upload'),
```

Comment:
* Bad URL (make it context-relevant)
* random view function name (make it a class too)
* "csv_upload" is also random, change to follow our style
* You added whitespaces